save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/service/DockerService.kt

```diff
@@ -114,7 +117,22 @@ class DockerService(
             .toBodilessEntity()
             .subscribe()
         agentRunner.start(execution.id!!)
-        log.info("Successfully started all containers for execution.id=$executionId")
+        log.info("Made request to start containers for execution.id=$executionId")
+        // Check, whether the agents were actually started, if yes, all cases will be covered by themselfs and HeartBeatInspector,
+        // if no, mark execution as failed with internal error here
+        val now = Clock.System.now()
+        var duration = 0L
+        while (duration < AGENTS_START_TIMEOUT && !areAgentsHaveStarted.get()) {
+            Thread.sleep(AGENTS_START_SLEEP_INTERVAL)
```

Comment:
Let's chain a `Mono.delay` after request to the backend instead of blocking a thread

Reply:
Done